addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
Ah, my bad, I thought it was just the Promise resolution it waited for. Been a while since I looked at that code 😅 

Either way though, the point I'm trying to make is that this may not be guaranteed. It works for now, definitely, and it looks like we're actually relying on that behavior in the tests for `vertical-collection`. The other option would be to send the event synchronously using `dispatchEvent`, like I mentioned before, and I'm not sure which one would be better here.